asmcomp/spacetime_profiling.ml

```diff
@@ -31,7 +31,12 @@ let something_was_instrumented () =
 let next_index_within_node ~part_of_shape ~label =
   let index = !index_within_node in
   begin match part_of_shape with
-  | Mach.Direct_call_point _ | Mach.Indirect_call_point ->
+  | Mach.Direct_call_point _ ->
+    incr index_within_node;
+    if Config.spacetime_call_counts then begin
+      incr index_within_node
+    end
```

Comment:
To guard against if-then-else errors.  I put them everywhere now unless the conditional is straightforward and fits entirely on a single line.